users/serializers.py

```diff
@@ -141,9 +145,12 @@ def validate_username(self, value):
         if not re.fullmatch(USERNAME_RE, trimmed_value):
             raise serializers.ValidationError(USERNAME_ERROR_MSG)
 
-        openedx_username_taken = False
+        openedx_validation_msg = ""
         try:
-            openedx_username_taken = check_username_exists_in_edx(trimmed_value)
+            openedx_validation_msg = validate_username_with_edx(trimmed_value)
+            openedx_validation_msg = OPENEDX_USERNAME_VALIDATION_MSGS_MAP.get(
```

Comment:
So I was thinking. If edX returns some error message for the username while validating account registration and the message is not related to the existing username that we had in `OPENEDX_USERNAME_VALIDATION_MSGS_MAP` we will get None for `openedx_validation_msg` even though there can be some other error for username?

In the above case, I think the user creation would still succeed, whereas I think if we get any error for the username field we should block the registration process, and in this case, we can send a generic message to the frontend for the user that the registration can't proceed.

Reply:
I think it would be good to take @rhysyngsun's opinion on this too.

Reply:
I have used the `.get()` method of the dictionary and provided the msg returned by edX as default. So, in case the msg does not exist in our `OPENEDX_USERNAME_VALIDATION_MSGS_MAP` then it will return the msg returned by edX as I am doing `OPENEDX_USERNAME_VALIDATION_MSGS_MAP.get(openedx_validation_msg, openedx_validation_msg)`

Reply:
The approach here is fine to pass through the openedx error message.